tests/harness/model/config.py

```diff
@@ -0,0 +1,129 @@
+
+        Args:
+            config: The configuration to merge into this one.
+        """
+
+        def check_duplicate_keys(
+            entries: Sequence[BaseModel], key_attr: str
+        ) -> List[str]:
+            """Checks for duplicated keys.
+
+            Args:
+                entries: List of pydantic objects.
+                key_attr: The attribute to use as key.
+
+            Returns:
+                A list of duplicate keys.
+            """
+            keys = [getattr(entry, key_attr) for entry in entries]
+            return [key for key in keys if keys.count(key) > 1]
+
+        self.deployments += config.deployments
+        self.secrets += config.secrets
+        self.tests += config.tests
+        self.requirements += config.requirements
+        self.environments += config.environments
+
+        duplicates = check_duplicate_keys(self.deployments, "name")
+        if duplicates:
+            raise ValueError(
+                f"Configuration error: deployments with duplicate "
+                f"names loaded from configuration file `{self._config_file}`: "
+                f"{', '.join(duplicates)}"
+            )
+        duplicates = check_duplicate_keys(self.secrets, "name")
+        if duplicates:
+            raise ValueError(
+                f"Configuration error: secrets with duplicate "
+                f"names loaded from configuration file `{self._config_file}`: "
+                f"{', '.join(duplicates)}"
+            )
+        duplicates = check_duplicate_keys(self.tests, "module")
+        if duplicates:
+            raise ValueError(
+                f"Configuration error: test requirements entries with "
+                f"duplicate module names loaded from configuration file "
+                f"`{self._config_file}`: {', '.join(duplicates)}"
+            )
+        duplicates = check_duplicate_keys(self.requirements, "name")
+        if duplicates:
+            raise ValueError(
+                f"Configuration error: global requirements entries with "
+                f"duplicate names loaded from configuration file "
+                f"`{self._config_file}`: {', '.join(duplicates)}"
+            )
+        duplicates = check_duplicate_keys(self.environments, "name")
+        if duplicates:
+            raise ValueError(
+                f"Configuration error: environments with duplicate "
+                f"names loaded from configuration file "
+                f"`{self._config_file}`: {', '.join(duplicates)}"
+            )
+
+    def compile(self, harness: "TestHarness") -> None:
+        """Validates and compiles the configuration.
+
+        Args:
+            harness: The test harness to validate against.
+        """
+        list(map(lambda s: s.compile(harness), self.secrets))
+        list(map(lambda d: d.compile(harness), self.deployments))
+        list(
+            map(
+                lambda r: r.compile(harness),
```

Comment:
same here, it's essentially the same for all attributes so why not just loop over them

Reply:
Done